app/controllers/user_controller.py

```diff
@@ -0,0 +1,19 @@
+from typing import Any
+from flask import Blueprint, render_template, session
+from app.models import User
+from app.utils.session_decorators import session_required
+
+
+user_routes = Blueprint('user_routes', __name__)
+
+
+@user_routes.route('/<username>', methods=['GET'])
+@session_required
+def get_user(username) -> Any:
+    user = User.get_by_username(username)
+    if user:
+        is_me = user.user_id == session['token']
```

Comment:
and then here instead of accessing the session token directly
`is_me = user.user_id == user_id`

also just for future reference you dont want to access dicts with bracket notion unless you have already checked the key is in the dict or it will throw a KeyError if it isn't

Reply:
where do you get `user_id` from?

Reply:
I see!

Reply:
You can just pass it in as a parameter like i showed above `def get_user(user_id, username) -> Any:` the decorator i wrote `@session_required` will inject it in as the first parameter